src/fides/api/task/conditional_dependencies/evaluator.py

```diff
@@ -13,19 +14,75 @@
     Operator,
 )
 
+# Define operator methods for validation
 operator_methods = {
     Operator.exists: lambda a, _: a is not None,
     Operator.not_exists: lambda a, _: a is None,
     Operator.eq: py_operator.eq,
     Operator.neq: py_operator.ne,
-    Operator.lt: lambda a, b: a < b if a is not None else False,
-    Operator.lte: lambda a, b: a <= b if a is not None else False,
-    Operator.gt: lambda a, b: a > b if a is not None else False,
-    Operator.gte: lambda a, b: a >= b if a is not None else False,
+    Operator.lt: lambda a, b: (
+        a < b if a is not None and isinstance(a, numbers.Number) else False
+    ),
+    Operator.lte: lambda a, b: (
+        a <= b if a is not None and isinstance(a, numbers.Number) else False
+    ),
+    Operator.gt: lambda a, b: (
+        a > b if a is not None and isinstance(a, numbers.Number) else False
+    ),
+    Operator.gte: lambda a, b: (
+        a >= b if a is not None and isinstance(a, numbers.Number) else False
+    ),
     Operator.list_contains: lambda a, b: b in a if isinstance(a, list) else False,
-    Operator.not_in_list: lambda a, b: a not in b if isinstance(b, list) else True,
+    Operator.not_in_list: lambda a, b: b not in a if isinstance(a, list) else False,
+    Operator.starts_with: lambda a, b: (
+        a.startswith(b) if isinstance(a, str) and isinstance(b, str) else False
+    ),
+    Operator.ends_with: lambda a, b: (
+        a.endswith(b) if isinstance(a, str) and isinstance(b, str) else False
+    ),
+    Operator.contains: lambda a, b: (
+        b in a if isinstance(a, str) and isinstance(b, str) else False
+    ),
 }
 
+# Common operators that work with most data types
+COMMON_OPERATORS = {
+    Operator.eq,
+    Operator.neq,
+    Operator.exists,
+    Operator.not_exists,
+}
+
+# Numeric comparison operators
+NUMERIC_OPERATORS = {Operator.lt, Operator.lte, Operator.gt, Operator.gte}
+
+# String-specific operators
+STRING_OPERATORS = {
+    Operator.contains,
+    Operator.starts_with,
+    Operator.ends_with,
+    Operator.list_contains,
```

Comment:
Is list contains strictly a string operator? I imagine we would want to check if a numeric value is present in a numeric array